concordium-consensus/src/Concordium/KonsensusV1/TreeState/LowLevel/LMDB.hs

```diff
@@ -448,6 +484,79 @@ asWriteTransaction t = do
         (LMDB_Error _ _ (Right MDB_MAP_FULL)) -> Just ()
         _ -> Nothing
 
+-- |Helper function for implementing 'writeFinalizedBlocks'.
+writeFinalizedBlocksHelper ::
```

Comment:
We could simply pass in the pruned blocks (from the `PrunedResult` at the call site, and delete those to make this a bit simpler)
i.e. 

```suggestion
writeFinalizedBlocksHelper ::writeFinalizedBlocksHelper ::
    (HasDatabaseHandlers dbh pv, IsProtocolVersion pv) =>
    [BlockHash] ->
    [StoredBlock pv] ->
    FinalizationEntry ->
    dbh ->
    MDB_txn ->
    IO [BlockHash]
writeFinalizedBlocksHelper prunedBlocks finBlocks finEntry dbh txn = do    
    -- 1. Delete the pruned blocks from the blocks table. 
    -- 2. Delete the QC's which have a round less than or equal to succQC of the finalization entry.
    mapM_ (deleteRecord txn (dbh ^. blockStore)) prunedBlocks
    withCursor txn (dbh ^. nonFinalizedQuorumCertificateStore) $ \cursor -> do
        let loop' Nothing = return ()
            loop' (Just (Left e)) = throwM (DatabaseInvariantViolation e)
            loop' (Just (Right (rnd, _))) = do
                case compare rnd (qcRound $ feSuccessorQuorumCertificate finEntry) of
                    LT -> do
                        deleteAtCursor cursor
                        loop' =<< getCursor CursorNext cursor
                    GT -> do
                        deleteAtCursor cursor
                        loop' =<< getCursor CursorNext cursor
                    _ -> loop' Nothing                    
        loop' =<< getCursor CursorFirst cursor
```